.github/workflows/release.yaml

```diff
@@ -1,31 +1,37 @@
-# reusable workflow triggered manually
-name: Release charm to other tracks and channels
+name: Release to Charmhub
 
 on:
+  push:
+    branches:
+      - main
+      - track/**
   workflow_dispatch:
     inputs:
-      destination-channel:
-        description: 'Destination Channel'
-        required: true
-      origin-channel:
-        description: 'Origin Channel'
-        required: true
-      charm-name:
-        description: 'Charm subdirectory name'
-        required: true
+      destination_channel:
+        description: CharmHub channel to publish to
+        required: false
+        default: 'latest/edge'
+        type: string
```

Comment:
Transferred comment from https://github.com/canonical/kfp-operators/pull/655#discussion_r1920090546
> is this input ever used? I would imagine you would just dispatch the workflow on the correct branch and then the charming-actions/channel would get the channel right?

@carlcsaposs-canonical we do use this input in workflow dispatch, for example during our beta release we use it to release charms from `latest/edge` to `<version>/beta`

Reply:
yeah that's true, I think in this case there's no use for `workflow_dispatch` for this workflow. WDYT?

Reply:
done